src/app/systelab-components/datepicker/datepicker.component.ts

```diff
@@ -172,61 +172,66 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 
 			if (dateStr.length >= 2) {
 
+				let numDays = 0;
+				let numMonths = 0;
+				let numYears = 0;
+
 				if (dateStr.lastIndexOf('d') === dateStr.length - 1) {
 					const days: number = Number(dateStr.replace('d', '')
 						.replace('D', ''));
-					this.currentDate = new Date();
-					emit = false;
 					if (!isNaN(days)) {
-						this.currentDate.setDate(today.getDate() - days);
-						emit = true;
+						numDays = days;
 					}
 				} else if (dateStr.lastIndexOf('w') === dateStr.length - 1) {
 					const weeks: number = Number(dateStr.replace('w', '')
 						.replace('W', ''));
-					this.currentDate = new Date();
-					emit = false;
 					if (!isNaN(weeks)) {
-						this.currentDate.setDate(today.getDate() - (weeks * 7));
-						emit = true;
+						numDays = (weeks * 7);
 					}
 				} else if (dateStr.lastIndexOf('s') === dateStr.length - 1) {
 					const weeks: number = Number(dateStr.replace('s', '')
 						.replace('S', ''));
-					this.currentDate = new Date();
-					emit = false;
 					if (!isNaN(weeks)) {
-						this.currentDate.setDate(today.getDate() - (weeks * 7));
-						emit = true;
+						numDays =  (weeks * 7);
```

Comment:
I suggest to refactor the function changeDate() and split it in several functions

Reply:
Done